Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -172,6 +174,9 @@ class EdgeHitProcessor: HitProcessing {
             return nil
         }
 
+        let edgeEndpointStr = configSharedState[EdgeConstants.SharedState.Configuration.EDGE_ENVIRONMENT] as? String ?? EdgeEndpoint.production.rawValue
+        let edgeEndpoint = EdgeEndpoint(rawValue: edgeEndpointStr.lowercased()) ?? EdgeEndpoint.production
```

Comment:
I would prefer to have a default in EdgeEndpoint enum so we don't need to handle this case here, and we can pass nil, empty strings, random strings without worry.

Reply:
This isn't really possible unless we created our own initializer for the `EdgeEndpoint`. The default `rawValue` initializer that is provided for an enum with an associated type doesn't allow for `nil` argument and is also marked as a failable initializer.

I'd prefer not to overload the `rawValue` initializer to modify the behavior as it would become inconsistent with other enums in Swift and would be a net increase in overall lines of code.
https://developer.apple.com/documentation/swift/rawrepresentable/1538354-init

Reply:
We do that for consentStatus, shouldn't be that bad, right? 
The main advantage of having that contained in one place is that in case the code is updated in the EdgeHitProcessor, the logic for determining the correct endpoint doesn't need to be impacted.
The processor should just extract the value from the config ss, but the enum will take care of returning the correct endpoint for that value.